File: src/typescript/transaction/operations/update-proposal/update-proposal-end-date.ts
import { createHiveChain, UpdateProposalOperation } from '@hiveio/wax';

// Initialize hive chain interface
const chain = await createHiveChain();

// Initialize a transaction object
const tx = await chain.createTransaction();

const proposalId = 1;
const creator = "your-account";
const dailyPay = chain.hbdCoins(100); // 100.000 HBD
const subject = "Proposal Update";
const permlink = "proposal-update";
const endDate = '2023-03-14' // You can also give end date as a timestamp, eg. 1678917600000

tx.pushOperation(new UpdateProposalOperation({
    proposalId,
    creator,
    dailyPay,
    subject,
    permlink,
    endDate
}));

// Get a transaction object holding all operations and transaction TAPOS & expiration data, but transaction is **not signed yet**
console.log(tx.transaction);

File: src/rust/transaction/operations/update_proposal/update_proposal_end_date.rs
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize hive chain interface
    let chain = create_hive_chain(None)?;

    // Initialize a transaction object
    let mut tx = chain.create_transaction(None).await?;

    let proposal_id = 1;
    let creator = "your-account";
    let daily_pay = chain.hbd_coins(100)?; // 100.000 HBD
    let subject = "Proposal Update";
    let permlink = "proposal-update";
    let end_date = "2023-03-14T00:00:00".parse()?;

    tx.push_complex_operation(
        &chain,
        UpdateProposalOperation {
            proposal_id,
            creator: creator.into(),
            daily_pay: NaiAssetConvertible::Asset(daily_pay),
            subject: subject.into(),
            permlink: permlink.into(),
            end_date: Some(end_date),
        },
    )?;

    /*
    Get a transaction object holding all operations and transaction
    TAPOS & expiration data, but transaction is **not signed yet**
    */
    println!("{:#?}", tx.transaction());

    Ok(())
}
